app/packages/state/src/recoil/selectors.ts

```diff
@@ -384,7 +384,24 @@ export const similarityMethods = selector<{
 }>({
   key: "similarityMethods",
   get: ({ get }) => {
-    const methods = get(datasetAtom)?.brainMethods || [];
+    let methods = get(datasetAtom)?.brainMethods || [];
+    const isGroupDataset = get(isGroup);
+    const activeSlice = get(currentSlice(get(isModalActive)));
+
+    if (isGroupDataset && activeSlice) {
+      methods = methods.filter(({ viewStages }) => {
+        return viewStages.some((vs) => {
+          const { _cls, kwargs } = JSON.parse(vs);
+          if (_cls === "fiftyone.core.stages.SelectGroupSlices") {
+            const sliceValue = kwargs.filter(
+              (kwarg: string[]) => kwarg[0] === "slices"
+            )?.[0]?.[1];
+            if (sliceValue && sliceValue !== activeSlice) return false;
+          }
+          return true;
+        });
+      });
+    }
```

Comment:
@benjaminpkane @lanzhenw added this filters out a method if the activeSlice is not the one one with the similarity computed. 

Shows help message for the active slice without similarity computed

https://github.com/voxel51/fiftyone/assets/109545780/bb2100b0-68f1-41b7-992d-80c72b444f71